listenbrainz/webserver/static/js/src/metadata-viewer/MetadataViewer.tsx

```diff
@@ -156,14 +156,17 @@ export default function MetadataViewer(props: MetadataViewerProps) {
 
   const artistMBID = first(recordingData?.artist_mbids);
   const releaseMBID = recordingData?.release_mbid ?? metadata?.release?.mbid;
+  const CAAReleaseMBID = metadata?.release?.caa_release_mbid;
+  const CAAID = metadata?.release?.caa_id;
   let coverArtSrc = "/static/img/cover-art-placeholder.jpg";
-  if (releaseMBID) {
-    if (metadata?.release?.caa_id) {
-      coverArtSrc = `https://coverartarchive.org/release/${releaseMBID}/${metadata.release.caa_id}-500.jpg`;
-    } else {
-      // Backup if we don't have the CAA ID
-      coverArtSrc = `https://coverartarchive.org/release/${releaseMBID}/front`;
-    }
+  if (CAAReleaseMBID && CAAID) {
+    coverArtSrc = `https://coverartarchive.org/release/${CAAReleaseMBID}/${CAAID}-500.jpg`;
+  } else {
+    // Backup if we don't have the CAA ID.
+    // Try fetching using CAA release MBID and fall back on Release MBID
+    coverArtSrc = `https://coverartarchive.org/release/${
+      CAAReleaseMBID ?? releaseMBID
```

Comment:
This should be the other way around I think to prefer user submitted mbids.

Reply:
Since this is in the metadata lookup, the releaseMBID in question is our matched canonical release, right?
There's not user submitted data in this AFAIU, but maybe I'm missing something?

Reply:
Ah right, so that one is really of no use then. The backend will have already checked canonical releases. However, the user submitted mbid is available under `recordingData?.release_mbid` so maybe we can use that to get releaseMbid.

Reply:
OK, makes sens, we do already use it as first choice (`const releaseMBID = recordingData?.release_mbid ?? metadata?.release?.mbid;`) so as you suggested I'm inverting the order to prefer user-submitted data in that fallback: `releaseMBID ?? CAAReleaseMBID`